xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -2662,7 +2662,15 @@ void CActiveAE::LoadSettings()
   m_settings.resampleQuality = static_cast<AEQuality>(settings->GetInt(CSettings::SETTING_AUDIOOUTPUT_PROCESSQUALITY));
   m_settings.atempoThreshold = settings->GetInt(CSettings::SETTING_AUDIOOUTPUT_ATEMPOTHRESHOLD) / 100.0;
   m_settings.streamNoise = settings->GetBool(CSettings::SETTING_AUDIOOUTPUT_STREAMNOISE);
-  m_settings.silenceTimeout = settings->GetInt(CSettings::SETTING_AUDIOOUTPUT_STREAMSILENCE) * 60000;
+
+  // Audio keep alive timeout in minutes [0 - 10]
+  // but with setting "Always" minutes = 2147483647 (INT_MAX)
+  const int minutes = settings->GetInt(CSettings::SETTING_AUDIOOUTPUT_STREAMSILENCE);
+
+  if (minutes <= 10)
+    m_settings.silenceTimeout = minutes * 60000; // convert to ms
+  else
+    m_settings.silenceTimeout = 3 * 60 * 60000; // limit to 3h to avoid overflows
```

Comment:
Semantically this is not correct and I fear other places where this is an issue as well.

For the always value I would change the Logic. Imagine a user falling a sleep and after three hours his AVR makes the click noise :-)

Reply:
I can change this to:

```c++
else
  m_settings.silenceTimeout = INT_MAX;
```
which is ~596 hours in ms (enough for the user to sleep peacefully)